--- glib-2.44.0/gio/gwin32networking.h.orig	2015-03-20 18:33:38.000000000 +0100
+++ glib-2.44.0/gio/gwin32networking.h	2015-03-23 21:03:15.924279829 +0100
@@ -24,8 +24,9 @@
 /* Check if more ANSI-compliant Winsock2 functions are provided */
 /* For run-time compatibility with Windows XP, remove when XP support dropped */
 
-typedef INT (WSAAPI *PFN_InetPton) (INT, PCTSTR, PVOID);
-typedef PCTSTR (WSAAPI *PFN_InetNtop) (INT, PVOID, PTSTR, size_t);
+typedef ULONG NET_IFINDEX;
+typedef INT (WSAAPI *PFN_InetPton) (INT, LPCTSTR, PVOID);
+typedef LPCTSTR (WSAAPI *PFN_InetNtop) (INT, PVOID, PTSTR, size_t);
 typedef NET_IFINDEX (WINAPI *PFN_IfNameToIndex) (PCSTR);
 
 typedef struct _GWin32WinsockFuncs
--- glib-2.58.0/gio/gwin32appinfo.c.orig	2018-08-30 19:12:08.000000000 +0200
+++ glib-2.58.0/gio/gwin32appinfo.c	2018-09-18 20:29:20.206457756 +0200
@@ -22,6 +22,7 @@
 
 #include "config.h"
 
+#include <wctype.h>
 #include <string.h>
 
 #include "gcontenttype.h"
--- glib-2.46.0/gio/gwin32registrykey.c.orig	2015-09-21 12:44:58.000000000 +0200
+++ glib-2.46.0/gio/gwin32registrykey.c	2015-09-22 18:50:25.926459740 +0200
@@ -25,10 +25,15 @@
 #pragma warning ( disable:4005 )
 #endif
 #include <windows.h>
+#ifdef __MINGW32__
+#include <ddk/ntstatus.h>
+#include <ddk/ntddk.h>
+#else
 #include <ntstatus.h>
 #include <winternl.h>
+#endif
 
-#ifndef _WDMDDK_
+#if !defined(_WDMDDK_) && !defined(_DDK_WINDDK_H)
 typedef enum _KEY_INFORMATION_CLASS {
   KeyBasicInformation,
   KeyNodeInformation,
--- glib-2.58.0/glib/gstdio.c.orig	2018-08-30 19:12:08.000000000 +0200
+++ glib-2.58.0/glib/gstdio.c	2018-09-14 18:41:52.350477519 +0200
@@ -36,6 +36,9 @@
 #include <direct.h>
 #include <io.h>
 #include <sys/utime.h>
+#ifdef __MINGW32__
+#include <winioctl.h>
+#endif
 #else
 #include <utime.h>
 #include <errno.h>
@@ -54,6 +57,7 @@
 #endif
 
 #if defined (G_OS_WIN32)
+#  if !defined(__MINGW32__)
 
 /* We can't include Windows DDK and Windows SDK simultaneously,
  * so let's copy this here from MinGW-w64 DDK.
@@ -90,6 +94,7 @@
     } GenericReparseBuffer;
   };
 } REPARSE_DATA_BUFFER, *PREPARSE_DATA_BUFFER;
+#  endif
 
 static int
 w32_error_to_errno (DWORD error_code)
--- glib-2.58.0/glib/giowin32.c.orig	2018-08-30 19:12:08.000000000 +0200
+++ glib-2.58.0/glib/giowin32.c	2018-09-15 10:12:54.423173008 +0200
@@ -70,6 +70,7 @@
 #include "glib.h"
 
 #include <stdlib.h>
+#include <winerror.h>
 #include <winsock2.h>
 #include <windows.h>
 #include <conio.h>
--- glib-2.58.0/gio/gioerror.c.orig	2018-08-30 19:12:08.000000000 +0200
+++ glib-2.58.0/gio/gioerror.c	2018-09-17 20:04:51.387461074 +0200
@@ -23,6 +23,7 @@
 #include "gioerror.h"
 
 #ifdef G_OS_WIN32
+#include <winerror.h>
 #include <winsock2.h>
 #endif
 
--- glib-2.58.0/gio/gnetworking.h.in.orig	2018-08-30 19:12:08.000000000 +0200
+++ glib-2.58.0/gio/gnetworking.h.in	2018-09-18 19:33:44.179829189 +0200
@@ -29,6 +29,7 @@
 #include <winsock2.h>
 #include <ws2tcpip.h>
 #include <windns.h>
+#include <winerror.h>
 #include <mswsock.h>
 @WSPIAPI_INCLUDE@
 #include <iphlpapi.h>
--- glib-2.58.0/gio/gwin32networkmonitor.c.orig	2018-08-30 19:12:08.000000000 +0200
+++ glib-2.58.0/gio/gwin32networkmonitor.c	2018-09-26 17:22:59.000209749 +0200
@@ -28,6 +28,7 @@
 
 #include <winsock2.h>
 #include <ws2tcpip.h>
+#include <netioapi.h>
 #include <iphlpapi.h>
 #include <stdio.h>
 
