--- glib-2.32.4/gio/gcontenttype.c.orig	2012-07-14 22:33:18.000000000 +0200
+++ glib-2.32.4/gio/gcontenttype.c	2012-07-24 16:48:52.400116342 +0200
@@ -332,7 +332,7 @@
 
   /* our test suite and potentially other code used -1 in the past, which is
    * not documented and not allowed; guard against that */
-  g_return_val_if_fail (data_size != (gsize) -1, g_strdup (XDG_MIME_TYPE_UNKNOWN));
+  g_return_val_if_fail (data_size != (gsize) -1, g_strdup ("*"));
 
   if (filename)
     {
--- glib-2.34.0/gio/gregistrysettingsbackend.c.orig	2012-08-19 16:39:53.000000000 +0200
+++ glib-2.34.0/gio/gregistrysettingsbackend.c	2012-09-26 17:54:44.204962755 +0200
@@ -233,7 +233,7 @@
   if (result_code == ERROR_KEY_DELETED)
     trace ("(%s)", win32_message);
   else
-    g_message (win32_message);
+    g_message ("%s", win32_message);
 };
 
 
--- glib-2.34.2/gio/gsocket.c.orig	2012-11-10 04:27:29.000000000 +0100
+++ glib-2.34.2/gio/gsocket.c	2012-11-12 21:22:20.558859131 +0100
@@ -2391,7 +2391,7 @@
   g_return_val_if_fail (G_IS_SOCKET (socket), -1);
 
 #if defined(G_OS_WIN32)
-  if (WSAIoctl (socket->priv->fd, FIONREAD, NULL, 0, &avail, sizeof (avail), 0, 0) == SOCKET_ERROR)
+  if (WSAIoctl (socket->priv->fd, FIONREAD, NULL, 0, &avail, sizeof (avail), 0, NULL, NULL) == SOCKET_ERROR)
     return -1;
 #elif defined(SO_NREAD)
   if (getsockopt (socket->priv->fd, SOL_SOCKET, SO_NREAD, &avail, &avail_len) < 0)
