--- glib-2.34.0/gio/gregistrysettingsbackend.c.orig	2012-08-19 16:39:53.000000000 +0200
+++ glib-2.34.0/gio/gregistrysettingsbackend.c	2012-09-26 17:54:44.204962755 +0200
@@ -233,7 +233,7 @@
   if (result_code == ERROR_KEY_DELETED)
     trace ("(%s)", win32_message);
   else
-    g_message (win32_message);
+    g_message ("%s", win32_message);
 };
 
 
--- glib-2.40.0/glib/grand.c.orig	2014-02-22 16:29:07.000000000 +0100
+++ glib-2.40.0/glib/grand.c	2014-03-25 18:45:39.693348981 +0100
@@ -55,6 +55,7 @@
 #endif
 
 #ifdef G_OS_WIN32
+#include <windows.h>
 #include <stdlib.h>
 #endif
 
--- glib-2.44.0/gio/gwin32networking.h.orig	2015-03-20 18:33:38.000000000 +0100
+++ glib-2.44.0/gio/gwin32networking.h	2015-03-23 21:03:15.924279829 +0100
@@ -24,8 +24,9 @@
 /* Check if more ANSI-compliant Winsock2 functions are provided */
 /* For run-time compatibility with Windows XP, remove when XP support dropped */
 
-typedef INT (WSAAPI *PFN_InetPton) (INT, PCTSTR, PVOID);
-typedef PCTSTR (WSAAPI *PFN_InetNtop) (INT, PVOID, PTSTR, size_t);
+typedef ULONG NET_IFINDEX;
+typedef INT (WSAAPI *PFN_InetPton) (INT, LPCTSTR, PVOID);
+typedef LPCTSTR (WSAAPI *PFN_InetNtop) (INT, PVOID, PTSTR, size_t);
 typedef NET_IFINDEX (WINAPI *PFN_IfNameToIndex) (PCSTR);
 
 typedef struct _GWin32WinsockFuncs
