--- glib-2.14.0/glib/glib.symbols.orig	2007-08-03 19:52:48.000000000 +0200
+++ glib-2.14.0/glib/glib.symbols	2007-08-21 18:35:27.116366403 +0200
@@ -691,11 +691,13 @@
 g_slice_set_config
 g_slice_get_config
 g_slice_get_config_state
+#ifdef G_ENABLE_DEBUG
 #ifdef INCLUDE_INTERNAL_SYMBOLS
 g_slice_debug_tree_statistics
 #endif
 #endif
 #endif
+#endif
 
 #if IN_HEADER(__G_MESSAGES_H__)
 #if IN_FILE(__G_MESSAGES_C__)
